pkg/virt-controller/watch/pool/pool.go

```diff
@@ -889,9 +890,76 @@ func (c *Controller) opportunisticUpdate(pool *poolv1.VirtualMachinePool, vmOutd
 
 func (c *Controller) proactiveUpdate(pool *poolv1.VirtualMachinePool, vmUpdatedList []*virtv1.VirtualMachine) error {
 	var wg sync.WaitGroup
-	wg.Add(len(vmUpdatedList))
+
+	// Calculate maxUnavailable value (defaults to 25% if not specified)
+	maxUnavailable := intstr.FromString("25%")
+	if pool.Spec.MaxUnavailable != nil {
+		maxUnavailable = *pool.Spec.MaxUnavailable
+	}
+
+	// Calculate the maximum number of VMIs that can be unavailable
+	var maxUnavailableInt int
+	if maxUnavailable.Type == intstr.String {
+		// Handle percentage
+		percentage, err := strconv.ParseInt(strings.TrimSuffix(maxUnavailable.StrVal, "%"), 10, 32)
+		if err != nil {
+			return fmt.Errorf("invalid maxUnavailable percentage: %v", err)
+		}
+		maxUnavailableInt = int(float64(len(vmUpdatedList)) * float64(percentage) / 100.0)
+	} else {
+		// Handle absolute number
+		maxUnavailableInt = int(maxUnavailable.IntVal)
+	}
+
+	// Ensure at least one VMI can be unavailable
+	if maxUnavailableInt < 1 {
+		maxUnavailableInt = 1
+	}
+
+	// Count currently unavailable VMIs (those with ready condition false)
+	unavailableCount := 0
+	for _, vm := range vmUpdatedList {
+		vmiKey := controller.NamespacedKey(vm.Namespace, vm.Name)
+		obj, exists, _ := c.vmiStore.GetByKey(vmiKey)
+		if !exists {
```

Comment:
Isn't that going to be an issue? In that case we should try to get the VM with the client and eventually return with an error if it cannot be found?